### DISCUSSION TITLE
Typescript Accessibility

### DISCUSSION THREAD

**Discussion by Malewooo16** [saadeghi/daisyui#2436] [General]
It's really difficult to use components with functions such as modals with typescript, add types or documentations for typescript users.

<button className="btn" onClick={()=>document.getElementById('my_modal_2').showModal()}>open modal</button>

()=>document.getElementById('my_modal_2:  **Object is possibly null**
.showModal() **Property 'showModal' does not exist on type 'HTMLElement'.**

**Comment by victornasar** [comment]
Having the same issue here as well.

**Comment by saadeghi** [comment]
Ask this question on typescript repo. It's not related to daisyUI. `showModal` is a standard HTML method: https://developer.mozilla.org/en-US/docs/Web/API/HTMLDialogElement/showModal

**Comment by kriskratz** [comment]
Sorry to post in a closed thread. @Malewooo16 using your example, in Typescript just cast the element to HTMLDialogElement, like this:

`<button className="btn" onClick={()=>(document.getElementById('my_modal_2') as HTMLDialogElement).showModal()}>open modal`

That takes care of the errors for me.

**Reply by bobsbeenjamin** [reply]
So simple, yet so beautiful

**Reply by Bulbul-Programing** [reply]
thank you brother
